### PR TITLE
Simplify implementation.

### PR DESCRIPTION
**Pull request by saschadoemer**
Simplify the implementation and remove double-checking for Base64 needs. Since every type that is base64 encoded can be chunked it is only necessary to ask for the subset of chunkable messages.

### CODE REVIEW

File: agrirouter-sdk-java-impl/src/main/java/com/dke/data/agrirouter/impl/messaging/encoding/EncodeMessageServiceImpl.java [hunk 1]
```diff
@@ -106,7 +106,7 @@ public String encode(
     messageHeaderParameters.validate();
     payloadParameters.validate();
 
-    if (messageHeaderParameters.getTechnicalMessageType().getNeedsChunking()) {
+    if (messageHeaderParameters.getTechnicalMessageType().needsBase64Encoding()) {
```
Comment (Frank-Wiebeler) [inline agrirouter-sdk-java-impl/src/main/java/com/dke/data/agrirouter/impl/messaging/encoding/EncodeMessageServiceImpl.java:109]: I would like to add a comment here, e.g.
```
//Data that needs Base64Encoding also needs chunking. That's what we are checking here
if (messageHeaderParameters.getTechnicalMessageType().needsBase64Encoding()) {
```

**Review by Frank-Wiebeler — CHANGES_REQUESTED** [review]
The change request is really really small, when this is done a merge is OK.

Thank you for your great work. 👍

File: agrirouter-sdk-java-impl/src/main/java/com/dke/data/agrirouter/impl/messaging/encoding/EncodeMessageServiceImpl.java (hunk 1 above)
Comment (saschadoemer) [reply agrirouter-sdk-java-impl/src/main/java/com/dke/data/agrirouter/impl/messaging/encoding/EncodeMessageServiceImpl.java:109]: Inline comments are against clean code rules >>> http://www.laputan.org/pub/patterns/fowler/smells.pdf

I will find a better solution.

**Review by Frank-Wiebeler — APPROVED** [review]
Awesome! :)

**Pull request merged**